utils.go

```diff
@@ -74,6 +76,10 @@ func ScanForLinks(br io.Reader, cb func(cid.Cid)) error {
 				}
```

Comment:
Without knowing this code it is hard to know which functions are safe from returning io.EOFs, for example `CborReadHeaderBuf` above does not check for EOFs but from the names and arguments it seems like it could return them.  I don't know if this is actually what we should do but here and in `discard` it would help me be more sure EOF handling is correct if we did error escalation in a defer statement.